pkg/ccl/storageccl/export_storage.go

```diff
@@ -545,3 +546,36 @@ func (s *azureStorage) Delete(_ context.Context, basename string) error {
 func (s *azureStorage) Close() error {
 	return nil
 }
+
+// MakeHTTPFileServer returns a handler for GET/PUT/DELETE requests in `path`.
+func MakeHTTPFileServer(ctx context.Context, base string) http.HandlerFunc {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		defer r.Body.Close()
+		localfile := filepath.Join(base, filepath.Clean(filepath.FromSlash(r.URL.Path)))
+		switch r.Method {
+		case "PUT":
+			f, err := os.Create(localfile)
+			if err != nil {
+				log.Warning(ctx, errors.Wrap(err, "could not create requested path"))
+				http.Error(w, err.Error(), 500)
+				return
+			}
+			defer f.Close()
+			defer r.Body.Close()
```

Comment:
Hm, yeah, the above one probably isn't actually needed since only PUT requests expect to have a body actually... but I guess it can't hurt, so I'll cut this one.